src/cli/mod.rs

```diff
@@ -0,0 +1,241 @@
+            zstore: self.zstore,
+            limit: self.limit,
+            rc: self.rc,
+        }
+    }
+}
+
+fn get_field() -> Result<LanguageField> {
+    if let Ok(lurk_field) = std::env::var("LURK_FIELD") {
+        match lurk_field.to_lowercase().as_str() {
+            "bls12-381" => Ok(LanguageField::BLS12_381),
+            "pallas" => Ok(LanguageField::Pallas),
+            "vesta" => Ok(LanguageField::Vesta),
+            _ => bail!("Field not supported: {lurk_field}"),
+        }
+    } else {
+        Ok(DEFAULT_FIELD)
+    }
+}
+
+fn get_store<F: LurkField + for<'a> serde::de::Deserialize<'a>>(
+    zstore: &Option<PathBuf>,
+) -> Store<F> {
+    zstore
+        .as_ref()
+        .and_then(|zstore_path| fs::read(zstore_path).ok())
+        .and_then(|zstore_bytes| ZData::from_bytes(&zstore_bytes).ok())
+        .and_then(|zstore_data| from_z_data(&zstore_data).ok())
+        .map(|zstore: ZStore<F>| zstore.to_store())
+        .tap_none(|| {
+            if zstore.is_some() {
+                eprintln!("Failed to load ZStore. Starting with empty store.")
+            }
+        })
+        .unwrap_or_default()
+}
+
+macro_rules! new_repl {
+    ( $cli: expr, $field: path ) => {{
+        let limit = $cli.limit.unwrap_or(DEFAULT_LIMIT);
+        let rc = $cli.rc.unwrap_or(DEFAULT_RC);
+        let mut store = get_store(&$cli.zstore);
+        let env = store.nil();
+        Repl::<$field, Coproc<$field>>::new(store, env, limit, rc)?
+    }};
+}
+
+impl ReplCli {
+    pub fn run(&self) -> Result<()> {
+        macro_rules! repl {
+            ( $field: path ) => {{
+                let mut repl = new_repl!(self, $field);
+                if let Some(lurk_file) = &self.load {
+                    repl.load_file(lurk_file)?;
+                }
+                repl.start()
+            }};
+        }
+        match get_field()? {
+            LanguageField::Pallas => repl!(pallas::Scalar),
+            LanguageField::Vesta => repl!(vesta::Scalar),
+            LanguageField::BLS12_381 => repl!(blstrs::Scalar),
+        }
+    }
+}
+
+impl LoadCli {
+    pub fn run(&self) -> Result<()> {
+        macro_rules! load {
+            ( $field: path ) => {{
+                let mut repl = new_repl!(self, $field);
+                repl.load_file(&self.lurk_file)?;
+                if self.prove {
+                    repl.prove_last_claim()?;
+                }
+                Ok(())
+            }};
+        }
+        match get_field()? {
+            LanguageField::Pallas => load!(pallas::Scalar),
+            LanguageField::Vesta => load!(vesta::Scalar),
+            LanguageField::BLS12_381 => load!(blstrs::Scalar),
+        }
+    }
+}
+
+#[derive(Args, Debug)]
+struct VerifyArgs {
+    #[clap(value_parser)]
+    proof_file: PathBuf,
+}
+
+pub fn parse() -> Result<()> {
+    #[cfg(not(target_arch = "wasm32"))]
+    paths::create_lurk_dir()?;
+    if let Ok(repl_cli) = ReplCli::try_parse() {
+        repl_cli.run()
```

Comment:
I think this function should have a better name if it also runs the CLI after parsing.

Reply:
Done